quaistats/quaistats.go

```diff
@@ -688,610 +945,423 @@ func (s *Service) isJwtExpired(authJwt string) (bool, error) {
-		"id":        s.node,
-		"sideBlock": block,
-	}
-	report := map[string][]interface{}{
-		"emit": {"sideBlock", stats},
+		found = true
+
+		// Type assert the value to a *BatchObject
+		batchObject, ok = value.(*BatchObject)
+		if !ok {
+			log.Warn("Error casting value to *BatchObject")
+			return &tps{
+				TPS1m:                     totalTransactions1m / 60,
+				TPS1hr:                    totalTransactions1h / 3600,
+				TotalNumberTransactions1h: totalTransactions1h,
+			}
+		}
 	}
-	return conn.WriteJSON(report)
-}
 
-// reportBlock retrieves the current chain head and reports it to the stats server.
-func (s *Service) reportBlock(conn *connWrapper, block *types.Block) error {
-	// Gather the block details from the header or block chain
-	details := s.assembleBlockStats(block)
+	delta1hr := startBlockTime - batchObject.OldestBlockTime
 
-	// Assemble the block report and send it to the server
-	log.Trace("Sending new block to quaistats", "number", details.Number, "hash", details.Hash)
+	// Now use the BatchObject to get the oldest block time
+	TPS1hr := totalTransactions1h / delta1hr
+	TPS1m := totalTransactions1m / 60
 
-	if conn == nil || conn.conn == nil {
-		log.Warn("block connection is nil")
-		return errors.New("block connection is nil")
-	}
+	log.Trace("Generated tx stats", "tps1m", TPS1m, "tps1hr", TPS1hr, "totalTransactions1h", totalTransactions1h, "totalTransactions1m", totalTransactions1m, "oldest1hBlockTime", batchObject.OldestBlockTime, "startBlockTime", startBlockTime, "difference1h", delta1hr, "difference1m", 60, "startBlockNum", uint64(startBlockNum))
 
-	stats := map[string]interface{}{
-		"id":    s.node,
-		"block": details,
-	}
-	report := map[string][]interface{}{
-		"emit": {"block", stats},
+	// Now totalTransactions1h and totalTransactions1m have the transaction counts for the last c_blocksPerHour and c_txBatchSize blocks respectively
+	return &tps{
+		TPS1m:                     TPS1m,
+		TPS1hr:                    TPS1hr,
+		TotalNumberTransactions1h: totalTransactions1h,
 	}
-	return conn.WriteJSON(report)
 }
 
-// assembleBlockStats retrieves any required metadata to report a single block
-// and assembles the block stats. If block is nil, the current head is processed.
-func (s *Service) assembleBlockStats(block *types.Block) *blockStats {
-	// Gather the block infos from the local blockchain
-	var (
-		header  *types.Header
-		entropy *big.Int
-		txs     []txStats
-		uncles  []*types.Header
-	)
-	header = block.Header()
-	entropy = s.backend.TotalLogS(block.Header())
+func min(a uint64, b uint64) uint64 {
+	if a < b {
+		return a
+	}
+	return b
+}
 
-	txs = make([]txStats, len(block.Transactions()))
-	for i, tx := range block.Transactions() {
-		txs[i].Hash = tx.Hash()
+func (s *Service) assembleBlockDetailStats(block *types.Block) *blockDetailStats {
+	if block == nil {
+		log.Error("Block is nil")
+		return nil
 	}
-	uncles = block.Uncles()
+	header := block.Header()
+	location := header.NumberArray()
```

Comment:
why is this called location? you can index the NumberArray(0) directly I thin, no need for extra variables